deepspeed/module_inject/fusedqkv_utils.py

```diff
@@ -123,3 +123,54 @@ def _transpose_fused_qkvw(src, mp_size, fused_qkv_type=None, module=None):
     warning_once(f"Unrecognized fusedkqv weight type, default to using bloom type,"
                  f"please check in prepare_tp_fused_qkvw() to avoid potential calculation errors")
     return _bloom_type_transpose(src, mp_size)
+
+
+def shard_value_with_share_qk(
```

Comment:
Comments here needed (with an example) to help understand functionality of `shard_value_with_share_qk()`